apps/web/src/pages/AnnouncementsPage/SitewideAnnouncementForm.tsx

```diff
@@ -0,0 +1,216 @@
+  titleEn: string;
+  messageEn: string;
+  titleFr: string;
+  messageFr: string;
+};
+
+const apiDataToFormValues = (
+  apiData: SitewideAnnouncement | null | undefined,
+): FormValues => ({
+  isEnabled: !!apiData?.isEnabled,
+  publishDate: apiData?.publishDate ?? nowUTCDateTime(),
+  expiryDate: apiData?.expiryDate ?? nowUTCDateTime(),
+  titleEn: apiData?.title.en ?? "",
+  messageEn: apiData?.message.en ?? "",
+  titleFr: apiData?.title.fr ?? "",
+  messageFr: apiData?.message.fr ?? "",
+});
+
+const formValuesToApiData = (formValues: FormValues): SitewideAnnouncement => ({
+  isEnabled: formValues.isEnabled,
+  publishDate: formValues.publishDate,
+  expiryDate: formValues.expiryDate,
+  title: {
+    en: formValues.titleEn,
+    fr: formValues.titleFr,
+  },
+  message: {
+    en: formValues.messageEn,
+    fr: formValues.messageFr,
+  },
+});
+
+interface SitewideAnnouncementFormProps {
+  initialData: SitewideAnnouncement | null | undefined;
+  onUpdate: (data: SitewideAnnouncementInput) => Promise<void>;
+  onOpenChange: (isOpen: boolean) => void;
+  isSubmitting: boolean;
+}
+
+const SitewideAnnouncementForm = ({
+  initialData,
+  onUpdate,
+  onOpenChange,
+  isSubmitting,
+}: SitewideAnnouncementFormProps) => {
+  const intl = useIntl();
+  const methods = useForm<FormValues>({
+    defaultValues: apiDataToFormValues(initialData),
+  });
+  const { handleSubmit } = methods;
+
+  const handleSave = async (formValues: FormValues) => {
+    return onUpdate(formValuesToApiData(formValues))
+      .then(() => {
+        methods.reset(formValues, {
+          keepDirty: false,
+        });
+        onOpenChange(false);
+      })
+      .catch(() => methods.reset(formValues));
+  };
+
+  const validateDateTimeInput = (value: string | null): ValidateResult => {
+    if (!value) {
+      return intl.formatMessage(invalidDateTimeMessage);
+    }
+    try {
+      const parsedValue = formDateTimeStringToDate(value);
+      if (Number.isNaN(parsedValue.getTime())) {
+        return intl.formatMessage(invalidDateTimeMessage);
+      }
+      return undefined;
+    } catch {
+      return intl.formatMessage(invalidDateTimeMessage);
+    }
+  };
+
+  return (
+    <FormProvider {...methods}>
+      <form
+        onSubmit={handleSubmit(handleSave)}
+        data-h2-display="base(grid)"
+        data-h2-gap="base(x1)"
+        data-h2-grid-template-columns="base(repeat(2, 1fr))"
+      >
+        <div data-h2-grid-column="base(span 2)">
+          <SwitchInput
+            id="isEnabled"
+            name="isEnabled"
+            label={intl.formatMessage(labels.isEnabled)}
+          />
+        </div>
+        <Input
+          id="publishDate"
+          label={intl.formatMessage(labels.publishDateUtc)}
+          name="publishDate"
+          type="text"
+          rules={{
+            required: intl.formatMessage(errorMessages.required),
+            validate: validateDateTimeInput,
+          }}
+        />
```

Comment:
I was wondering why these were text fields but forgot our date picker does not support time 😢 . Oh well, this should only be used by a very small number of people so this should work 👍

Reply:
Yeah, this is clunky.  It would be nice if our component could handle time.